jcli/src/jcli_app/certificate/sign.rs

```diff
@@ -97,25 +108,79 @@ pub(crate) fn committee_vote_tally_sign(
+    if keys_str.len() > 1 {
+        return Err(Error::ExpectingOnlyOneSigningKey {
+            got: keys_str.len(),
+        });
+    }
+    let private_key = parse_ed25519_secret_key(keys_str[0].trim())?;
+    let id = private_key.to_public().as_ref().try_into().unwrap();
 
-            let signature = SingleAccountBindingSignature::new(&builder.get_auth_data(), |d| {
-                private_key.sign_slice(&d.0)
-            });
+    let signature = SingleAccountBindingSignature::new(&builder.get_auth_data(), |d| {
+        private_key.sign_slice(&d.0)
+    });
 
-            let proof = TallyProof::Public { id, signature };
+    let proof = match vote_tally.tally_type() {
+        PayloadType::Public => TallyProof::Public { id, signature },
+        // TODO: vote tally can only be public, add a proper error here
+        PayloadType::Private => return Err(Error::ExpectedSignedOrNotCertificate),
+    };
+    Ok(SignedCertificate::VoteTally(vote_tally, proof))
+}
 
-            Ok(SignedCertificate::VoteTally(vote_tally, proof))
-        }
+pub(crate) fn committee_encrypted_vote_tally_sign(
+    vote_tally: EncryptedVoteTally,
+    keys_str: &[String],
+    builder: TxBuilderState<SetAuthData<EncryptedVoteTally>>,
+) -> Result<SignedCertificate, Error> {
+    use chain_impl_mockchain::vote::PayloadType;
+
+    if keys_str.len() > 1 {
+        return Err(Error::ExpectingOnlyOneSigningKey {
+            got: keys_str.len(),
+        });
     }
+    let private_key = parse_ed25519_secret_key(keys_str[0].trim())?;
+    let id = private_key.to_public().as_ref().try_into().unwrap();
+
+    let signature = SingleAccountBindingSignature::new(&builder.get_auth_data(), |d| {
+        private_key.sign_slice(&d.0)
+    });
+
+    let proof = EncryptedVoteTallyProof { id, signature };
+    Ok(SignedCertificate::EncryptedVoteTally(vote_tally, proof))
+}
+
+pub(crate) fn committee_private_vote_tally_sign(
+    vote_tally: PrivateVoteTally,
+    keys_str: &[String],
+    builder: TxBuilderState<SetAuthData<PrivateVoteTally>>,
+) -> Result<SignedCertificate, Error> {
+    use chain_impl_mockchain::vote::PayloadType;
+
+    if keys_str.len() > 1 {
+        return Err(Error::ExpectingOnlyOneSigningKey {
+            got: keys_str.len(),
+        });
+    }
+    let private_key = parse_ed25519_secret_key(keys_str[0].trim())?;
+    let id = private_key.to_public().as_ref().try_into().unwrap();
+
+    let signature = SingleAccountBindingSignature::new(&builder.get_auth_data(), |d| {
+        private_key.sign_slice(&d.0)
+    });
+
+    let proof = match vote_tally.tally_type() {
+        // TODO: vote tally can only be private, add a proper error here
+        PayloadType::Public => return Err(Error::ExpectedSignedOrNotCertificate),
+        // TODO: how to pass the shares here...?
+        PayloadType::Private => TallyProof::Private {
+            id,
+            signature,
+            shares: vec![],
+        },
```

Comment:
I'm not sure how to spread the shares here. Did we talk about this already? I have quite a mix of certificates in my head now.